console/src/components/Services/Data/TablePermissions/molecules/PermButtonsSection.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback } from 'react';
+import Button from '../../../../Common/Button';
+import {
+  isJsonString,
+  getConfirmation,
+} from '../../../../Common/utils/jsUtils';
+import { FilterState } from '../utils';
+import { showErrorNotification } from '../../../Common/Notification';
+import { permChangePermissions, permChangeTypes } from '../Actions';
+import styles from '../../../../Common/Permissions/PermissionStyles.scss';
+
+interface PermButtonSectionProps {
+  readOnlyMode: string;
+  query: string;
+  localFilterString: FilterState;
+  dispatch: (...p: any) => void;
+  permissionsState: FilterState;
+  permsChanged: string;
+  currQueryPermissions: string;
+}
+
+const PermButtonSection: React.FC<PermButtonSectionProps> = ({
+  readOnlyMode,
+  query,
+  localFilterString,
+  dispatch,
+  permissionsState,
+  permsChanged,
+  currQueryPermissions,
+}) => {
+  if (readOnlyMode) {
+    return null;
+  }
+
+  const dispatchSavePermissions = useCallback(() => {
+    console.error(localFilterString);
+    const isInvalid = Object.values(localFilterString).some(val => {
+      if (val && !isJsonString(val)) {
+        return true;
+      }
+      return false;
+    });
+
+    if (isInvalid) {
+      dispatch(
+        showErrorNotification(
+          'Saving permissions failed',
+          'Row permission is not a valid JSON'
+        )
+      );
+      return;
+    }
+
+    dispatch(permChangePermissions(permChangeTypes.save));
+  }, [readOnlyMode, query, localFilterString, dispatch]);
+
+  const dispatchRemoveAccess = useCallback(() => {
+    const confirmMessage =
+      'This will permanently delete the currently set permissions';
+    const isOk = getConfirmation(confirmMessage);
+    if (isOk) {
+      dispatch(permChangePermissions(permChangeTypes.delete));
+    }
+  }, []);
+
+  const disableSave =
+    permissionsState.applySamePermissions.length !== 0 || !permsChanged;
+  const disableRemoveAccess = !currQueryPermissions;
```

Comment:
Does it need an additional variable?

Reply:
I saw this pattern used throughout for readabililty and thought of keeping it as it is. 
Changing now :)